src/importexport/imagesexport/internal/pdfwriter.cpp

```diff
@@ -147,11 +149,107 @@ Ret PdfWriter::writeList(const INotationPtrList& notations, io::IODevice& destin
     return true;
 }
 
-void PdfWriter::preparePdfWriter(QPdfWriter& pdfWriter, const QString& title, const QSizeF& size) const
+// Extract project metadata from current project or notation fallback
+ProjectMeta PdfWriter::getProjectMetadata(INotationPtr notation) const
+{
+    auto project = globalContext()->currentProject();
```

Comment:
It would be great if we could avoid the dependency on `globalContext` here. The notation being exported should not necessarily correspond to the currently opened project. 

I see two ways:
* Move the `metaInfo` method to `INotation` (and perhaps let `NotationProject` still have a `metaInfo` that just calls `notation->metaInfo`). 
* Give `INotation` a pointer to its `NotationProject`, and retrieve the project from the `notation`. Perhaps that's a good idea anyway, and shouldn't be difficult. Note that it should be a raw pointer or a weak pointer, but not a shared pointer, because otherwise the MasterNotation and NotationProject will keep each other alive (a "retain cycle", which is a kind of memory leak).